src/compiler/crystal/tools/doc/generator.cr

```diff
@@ -67,12 +67,18 @@ class Crystal::Doc::Generator
     end
 
     if filename
-      body = doc(program_type, File.read(filename))
+      raw_body = File.read(filename)
+      body = doc(program_type, raw_body)
     else
+      raw_body = ""
       body = ""
     end
 
     File.write "#{@dir}/index.html", MainTemplate.new(body, types, repository_name)
+
+    main_index = Main.new(raw_body, Type.new(self, @program), repository_name)
+    File.write "#{@dir}/index.json", main_index
+    File.write "#{@dir}/index.jsonp", main_index.to_jsonp
```

Comment:
We have many restrictions as long as using "file://" protocol and I don't think it should be supported. We can use `ruby -run -e httpd`, `python -mSimpleHTTPServer` or `python -mhttp.server` to serve local files.